internal/adminapi/client.go

```diff
@@ -76,13 +91,30 @@ func (c *Client) GetKongVersion(ctx context.Context) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed fetching Kong client root: %w", err)
 	}
-	version, ok := rootConfig["version"].(string)
-	if !ok {
-		return "", errors.New("malformed Kong version found in Kong client root")
+
+	versionKey := "version"
+	version, err := extractStringFromRoot(rootConfig, versionKey)
+	if err != nil {
+		return "", fmt.Errorf("malformed Kong version found in Kong client root: %w", err)
 	}
+
 	return version, nil
 }
 
+func extractStringFromRoot(data map[string]interface{}, key string) (string, error) {
+	val, ok := data[key]
+	if !ok {
+		return "", fmt.Errorf("%q key not found", key)
+	}
+
+	valStr, ok := val.(string)
+	if !ok {
+		return "", fmt.Errorf("%q key is not a string, actual type: %T", key, val)
```

Comment:
nit: `%q` might not work with all types 🤔 I haven't checked but I think we might be better off with `%v` here

Reply:
It's fine as `key` (that matches the `%q` placeholder) is a string.

Reply:
👍